packages/vite/src/buildFeServer.ts

```diff
@@ -41,85 +40,45 @@ export const buildFeServer = async ({ verbose, webDir }: BuildOptions = {}) => {
     )
   }
 
-  if (rwConfig.experimental?.rsc?.enabled) {
+  if (rscBuild) {
     if (!rwPaths.web.entries) {
       throw new Error('RSC entries file not found')
     }
 
-    await buildRscFeServer({
-      viteConfigPath,
-      webHtml: rwPaths.web.html,
-      entries: rwPaths.web.entries,
-      webDist: rwPaths.web.dist,
-      webDistServer: rwPaths.web.distServer,
-      webDistServerEntries: rwPaths.web.distServerEntries,
-    })
-
-    // Write a route manifest
-    return await buildRouteManifest()
-
-    //
-    // RSC specific code ends here
-    //
+    await buildRscClientAndWorker()
   }
 
-  //
-  // SSR Specific code below
-  //
+  // We generate the RSC client bundle in the buildRscFeServer function
+  // Streaming and RSC client bundles are **not** the same
+  if (streamingBuild && !rscBuild) {
+    console.log('Building client for streaming SSR...\n')
+    await buildWeb({ verbose })
+  }
+
+  // Generates the output used for the server (streaming/ssr but NOT rsc)
+  await buildForServer({ verbose })
+
+  await buildRouteHooks(verbose, rwPaths)
 
-  // Step 1A: Generate the client bundle
-  await buildWeb({ verbose })
+  // Write a route manifest
+  await buildRouteManifest()
+}
+
+async function buildForServer({ verbose = false }: { verbose?: boolean }) {
+  console.log('Starting server build.... \n')
+  const rwPaths = getPaths()
 
-  // Step 1B: Generate the server output
   await viteBuild({
-    configFile: viteConfigPath,
+    configFile: rwPaths.web.viteConfig as string,
```

Comment:
Why the type cast here?

Reply:
Because viteConfig is _technically_ still optional in redwood. Until we drop webpack support completely!

Reply:
> Because viteConfig is technically still optional in redwood. Until we drop webpack support completely!

Yeah. Can't wait to get rid of all the webpack stuff!

> But at this point in the code, it's guaranteed because of checks further up the tree!

I'd prefer an explicit check and a `throw`. Whenever I see `as` I get an itch 🐜